cognite/dm_clients/config.py

```diff
@@ -1,3 +1,5 @@
+from typing import TypedDict, cast
```

Comment:
Any reason not to use `pydantic` here? Will it not be good with validation that the user has provided settings correctly?

Reply:
Removed type hints, maybe in the future.